test/Chainweb/Test/Pact/Checkpointer.hs

```diff
@@ -310,3 +326,409 @@ testCheckpointer loggers CheckpointEnv{..} dbState00 = do
+  blockenv01 <- restoreNew _cpeCheckpointerNew bh01 (Just hash00) >>= assertEitherSuccess "restore failed"
+  withBlockEnv blockenv01 $ \benv -> do
+    addKeyset "k2" (KeySet [] (Name ">=" (Info Nothing))) benv
+    runBlockEnv benv $ do
+      keysetsTable <- callDb $
+        \db -> qry_ db ("SELECT * FROM [" <> domainTableName KeySets <> "];")
+               [RText, RInt, RInt, RInt, RBlob]
+      liftIO $ assertEqual "keyset table" keysetsTable ks2
+
+      liftIO $ saveNew _cpeCheckpointerNew bh01 hash01 >>= assertEitherSuccess "save failed"
+      runOnVersionTables
+        (\blocktable -> assertEqual "block table " blocktable bt01)
+        (\versionhistory -> assertEqual "version history " versionhistory vht01)
+        (\vtables -> assertEqual "versioned tables table " vtables vtt)
+
+  -- fork on blockheight = 1
+
+  let bh11 = BlockHeight 1
+
+  hash11 <- BlockHash <$> liftIO (merkleLogHash "0000000000000000000000000000001b")
+
+  blockenv11 <- restoreNew _cpeCheckpointerNew bh11 (Just hash00) >>= assertEitherSuccess "restore failed"
+
+  withBlockEnv blockenv11 $ \benv -> do
+
+    addKeyset "k1" (KeySet [] (Name ">=" (Info Nothing))) benv
+
+    runBlockEnv benv $ do
+      liftIO $ saveNew _cpeCheckpointerNew bh11 hash11 >>= assertEitherSuccess "save failed"
+      runOnVersionTables
+        (\blocktable -> assertEqual "block table " blocktable bt11)
+        (\versionhistory -> assertEqual "version history " versionhistory vht11)
+        (\vtables -> assertEqual "versioned tables table " vtables vtt)
+
+  where
+    vtt = []
+    ks1 = [[SText "k1",SInt 0,SInt 0,SInt 0,SBlob "{\"pred\":\">=\",\"keys\":[]}"]]
+    ks2 = [[SText "k1",SInt 0,SInt 0,SInt 0,SBlob "{\"pred\":\">=\",\"keys\":[]}"],[SText "k2",SInt 1,SInt 0,SInt 0,SBlob "{\"pred\":\">=\",\"keys\":[]}"]]
+    bt01 = [[SInt 0,SBlob "\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL"],[SInt 1,SBlob "0000000000000000000000000000001a"]]
+    vht01 = [[SInt 0,SInt 0],[SInt 0,SInt 1]]
+    bt11 = [[SInt 0,SBlob "\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL"],[SInt 1,SBlob "0000000000000000000000000000001b"]]
+    vht11 = [[SInt 0,SInt 0],[SInt 0,SInt 1],[SInt 1,SInt 1]]
+
+addKeyset :: KeySetName -> KeySet -> Method (BlockEnv SQLiteEnv) ()
+addKeyset = _writeRow chainwebpactdb Insert KeySets
+
+testRelational :: Assertion
+testRelational =
+  withTempSQLiteConnection []  $ \sqlenv -> do
+    let initBlockState = BlockState 0 Nothing (BlockVersion 0 0) M.empty (newLogger loggers "BlockEnvironment")
+        loggers = pactTestLogger False
+    msql <- newMVar (BlockEnv sqlenv initBlockState)
+    runBlockEnv msql initSchema
+    cpEnv <- initRelationalCheckpointerNew msql (newLogger loggers "RelationalCheckpointer") freeGasEnv
+    relationalTest cpEnv
+
+relationalTest :: CheckpointEnvNew SQLiteEnv -> Assertion
```

Comment:
add a comment that this copypastes the entire in-memory test.

Also we probably need to migrate the in-memory to match the `NewCheckpointer` in the subsequent PR. Having an in-memory version is useful for performance tests. Thus the comment should be a TODO to have a single test at that point.

Reply:
done